src/util.rs

```diff
@@ -65,6 +65,7 @@ impl LazyUsize {
             match self.0.compare_and_swap(Self::UNINIT, Self::ACTIVE, Relaxed) {
                 Self::UNINIT => {
                     let val = init();
+                    assert!(val != Self::ACTIVE);
```

Comment:
As I wrote earlier in #52 I would prefer to use code like this:
```rust
let mut val = init();
if val == Self::ACTIVE { val = Self::UNINIT }
self.0.store(val, Relaxed);
```

Reply:
Any particular reason why? Wouldn't having `init()` returning `ACTIVE` be a bug? Or are you concerned with the `LazyUsize` getting "stuck" in the `ACTIVE` state after a panic?

Reply:
Panicking pulls formatting machinery (unless some careful setup is done) and it results in a larger binary. I haven't thought about infinite loop (e.g. if user will catch the panic), but it's indeed another drawback.

I think we both agree that it's extremely unlikely that `init` will return `Self::ACTIVE`, so it does not really matter.

Reply:
I removed the assert and updated the docs. I used a slightly different implementation than above to maintain the invariant that `sync_init()` always returns the value of some run of `init()`.